File: 326-power-of-three/power-of-three.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfThree = function(n) {
    // 1 is always power of 3
    if(n==1) return true
    // 0 can cause infinte recursive loop
    if (n%3 != 0 || n < 3) {
        return false
    }
    let remainder = n/3
    // When decimal, not power of 3
    if (remainder % 1 != 0) {
        return false
    }
    if(remainder == 1) {
        return true
    }
    return isPowerOfThree(remainder)
};

File: 326-power-of-three/power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n==1:
            return True

        if n%3 != 0 or n < 3:
            return False

        remainder = n/3

        if remainder % 1 != 0:
            return False

        if remainder == 1:
            return True

        return self.isPowerOfThree(remainder)